rootfs/api/models/app.py

```diff
@@ -393,13 +393,18 @@ def _default_structure(self, release):
         if not release.build.sha:
             structure = {'cmd': 1}
 
-        # if a dockerfile exists without a procfile, assume docker workflow
-        elif release.build.dockerfile and not release.build.procfile:
+        elif release.build.procfile and 'web' in release.build.procfile:
```

Comment:
in the current branch logic if there's a dockerfile and a procfile, the process type will automatically be `{'cmd': 1}` because `elif release.build.dockerfile` will be evaluated before we even get to that point.

Good point though. Another pass at it:

```
elif release.build.procfile:
  if 'web' in release.build.procfile:
    structure = {'web': 1}
  else:
    # if a procfile exists without a web entry, assume heroku workflow
    # and return empty structure as only web type needs to be created by default and
    # other types have to be manually scaled
    structure = {}
    # this release has a procfile and a dockerfile, so we should assume the 'cmd'
    # process type in this case
    if release.build.dockerfile:
      structure = {'cmd': 1}
```